CMakeLists.txt

```diff
@@ -317,7 +317,16 @@ else()
   add_subdirectory(3party/freetype)
   add_subdirectory(3party/icu)
 endif()
-add_subdirectory(3party/jansson)
+
+set(JANSSON_EXAMPLES OFF)
+set(JANSSON_BUILD_DOCS OFF)
+set(JANSSON_WITHOUT_TESTS OFF)
+set(JANSSON_INSTALL OFF)
+find_package_or_fallback_to_3party(Jansson 2.14.0 3party/jansson/jansson/)
+if (NOT Jansson_FOUND)
+  include_directories("${PROJECT_BINARY_DIR}/3party/jansson/jansson/include")
```

Comment:
1. Why is this fallback needed?
2. If it is absolutely necessary, then why it can't be moved into the macro above?

Reply:
1. If the system doesn't provide `<jansson.h>`, then the out of tree build process for `3party/jansson/jansson/` makes this header from ` ${CMAKE_SOURCE_DIR}/3party/jansson/jansson/src/` available under  `${PROJECT_BINARY_DIR}/3party/jansson/jansson/include/` in build time, and by the time `cppjansson/jansson_handle.cpp` tries to include it, it does not appear to be among the include directories unless we explicitly request this.
2. Well the macro is meant to be generic and contain code that is applicable to most of the 3parties. This `include_directories` need appears to be rather specific to Jansson, at least so far.